setup.py

```diff
@@ -121,6 +122,121 @@
     'Topic :: Scientific/Engineering :: Visualization',
     ]
 
+
+class NoseTestCommand(TestCommand):
+    """Invoke unit tests using nose after an in-place build."""
+
+    description = "Invoke unit tests using nose after an in-place build."
+    user_options = [
+        ("pep8-only", None, "pep8 checks"),
+        ("omit-pep8", None, "Do not perform pep8 checks"),
+        ("nocapture", None, "do not capture stdout (nosetests)"),
+        ("nose-verbose", None, "be verbose (nosetests)"),
+        ("processes=", None, "number of processes (nosetests)"),
+        ("process-timeout=", None, "process timeout (nosetests)"),
+        ("with-coverage", None, "with coverage"),
+        ("detailed-error-msg", None, "detailed error message (nosetest)"),
+        ("tests=", None, "comma separated selection of tests (nosetest)"),
+    ]
+
+    def initialize_options(self):
+        self.pep8_only = None
+        self.omit_pep8 = None
+
+        # parameters passed to nose tests
+        self.processes = None
+        self.process_timeout = None
+        self.nose_verbose = None
+        self.nocapture = None
+        self.with_coverage = None
+        self.detailed_error_msg = None
+        self.tests = None
+
+    def finalize_options(self):
+        self.test_args = []
+        if self.pep8_only:
+            self.pep8_only = True
+        if self.omit_pep8:
+            self.omit_pep8 = True
+
+        if self.pep8_only and self.omit_pep8:
+            from distutils.errors import DistutilsOptionError
+            raise DistutilsOptionError(
+                "You are using several options for the test command in an "
+                "incompatible manner. Please use either one of --pep8-only,"
+                "--omit-pep8"
```

Comment:
should probably have a space here. It would render as "one of --pep8-only,--omit-pep8"